src/supautils.c

```diff
@@ -562,8 +564,70 @@ supautils_hook(PROCESS_UTILITY_PARAMS)
 		case T_AlterExtensionContentsStmt:
 			break;
 
+		/**
+		 * CREATE FOREIGN DATA WRAPPER <fdw>
+		 * ALTER FOREIGN DATA WRAPPER <fdw>
+		 */
+		case T_CreateFdwStmt:
+		case T_AlterFdwStmt:
+		{
+			if (superuser()) {
+				break;
+			}
+			if (privileged_role == NULL) {
+				break;
+			}
+			if (!OidIsValid(get_role_oid(privileged_role, true))) {
+				break;
+			}
+			if (GetUserId() != get_role_oid(privileged_role, false)) {
+				break;
+			}
+
+			switch_to_superuser(privileged_extensions_superuser);
+
+			run_process_utility_hook(prev_hook);
+
+			// GRANT USAGE on the FDW so we can do CREATE SERVER.
```

Comment:
The `privileged_role` won't be a superuser, so after the FDW is created, it [needs `GRANT USAGE`](https://www.postgresql.org/docs/current/sql-createserver.html#id-1.9.3.82.5) to do `CREATE SERVER ... FOREIGN DATA WRAPPER ...`

Reply:
One alternative I considered is to make the `privileged_role` the owner of the FDW, but since you need to be a superuser to own a FDW, I had to temporarily make the `privileged_role` a superuser - a bit icky, but might be a cleaner end state?

Reply:
Looks good. I see we can drop the FDW despite not being superuser later on.

Reply:
Ah nice catch, simplified the code a bit